codeguru_profiler_agent/sdk_reporter/sdk_reporter.py

```diff
@@ -90,11 +98,64 @@ def report(self, profile):
             )
             logger.info("Reported profile successfully")
             return True
+        except ClientError as error:
+            if error.response['Error']['Code'] == 'ResourceNotFoundException':
+                if self.is_lambda_one_click_integration_active():
+                    global is_create_pg_called_during_submit_profile
+                    is_create_pg_called_during_submit_profile = True
+                    self.create_pg_when_one_click_integration_is_active()
+            self._log_request_failed(operation="post_agent_profile", exception=error)
```

Comment:
> I would say to not log exception when we create the PG, but log something at INFO level that we are creating one PG.

Changed in the next commit.

Reply:
Because both ConfigureAgent and PostAgentProfile throw a ResourceNotFoundException. So if we fail to create a PG during the first refresh_configuration() call and the agent defaults to a default configuration, we can still attempt to create a PG during the report() call.